library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/OptionSelectorDetector.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder
+
+import android.view.ViewGroup
+import androidx.appcompat.widget.AppCompatSpinner
+
+internal class OptionSelectorDetector {
+    fun isOptionSelector(view: ViewGroup): Boolean {
+        val parentClassName = view.javaClass.canonicalName ?: ""
+        return parentClassName in OPTION_SELECTORS_CLASS_NAMES_SET ||
+            AppCompatSpinner::class.java.isAssignableFrom(view::class.java)
+    }
+
+    companion object {
+        private const val DROP_DOWN_LIST_CLASS_NAME = "android.widget.DropDownListView"
+        private const val APPCOMPAT_DROP_DOWN_LIST_CLASS_NAME =
```

Comment:
if we already have `androidx.appcompat` library in the runtime classpath, cannot we simply do something like `DropDownListView.javaClass == view.javaClass`? this will remove the need to depend on the name and have name entry in the Proguard rules.

Reply:
same as above